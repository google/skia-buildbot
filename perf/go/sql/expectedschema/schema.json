{
  "ColumnNameAndType": {
    "alerts.alert": "text def: nullable:YES",
    "alerts.config_state": "bigint def:0:::INT8 nullable:YES",
    "alerts.id": "bigint def:unique_rowid() nullable:NO",
    "alerts.last_modified": "bigint def: nullable:YES",
    "alerts.sub_name": "text def: nullable:YES",
    "alerts.sub_revision": "text def: nullable:YES",
    "anomalygroups.action": "text def: nullable:YES",
    "anomalygroups.action_time": "timestamp with time zone def: nullable:YES",
    "anomalygroups.anomaly_ids": "ARRAY def: nullable:YES",
    "anomalygroups.bisection_id": "text def: nullable:YES",
    "anomalygroups.common_rev_end": "bigint def: nullable:YES",
    "anomalygroups.common_rev_start": "bigint def: nullable:YES",
    "anomalygroups.creation_time": "timestamp with time zone def:now():::TIMESTAMPTZ nullable:YES",
    "anomalygroups.culprit_ids": "ARRAY def: nullable:YES",
    "anomalygroups.group_meta_data": "jsonb def: nullable:YES",
    "anomalygroups.id": "uuid def:gen_random_uuid() nullable:NO",
    "anomalygroups.last_modified_time": "timestamp with time zone def: nullable:YES",
    "anomalygroups.reported_issue_id": "text def: nullable:YES",
    "commits.author": "text def: nullable:YES",
    "commits.commit_number": "bigint def: nullable:NO",
    "commits.commit_time": "bigint def: nullable:YES",
    "commits.git_hash": "text def: nullable:NO",
    "commits.subject": "text def: nullable:YES",
    "culprits.anomaly_group_ids": "ARRAY def: nullable:YES",
    "culprits.group_issue_map": "jsonb def: nullable:YES",
    "culprits.host": "text def: nullable:YES",
    "culprits.id": "uuid def:gen_random_uuid() nullable:NO",
    "culprits.issue_ids": "ARRAY def: nullable:YES",
    "culprits.last_modified": "bigint def: nullable:YES",
    "culprits.project": "text def: nullable:YES",
    "culprits.ref": "text def: nullable:YES",
    "culprits.revision": "text def: nullable:YES",
    "favorites.id": "uuid def:gen_random_uuid() nullable:NO",
    "favorites.user_id": "text def: nullable:NO",
    "favorites.name": "text def: nullable:YES",
    "favorites.description": "text def: nullable:YES",
    "favorites.url": "text def: nullable:NO",
    "favorites.last_modified": "bigint def: nullable:YES",
    "graphsshortcuts.graphs": "text def: nullable:YES",
    "graphsshortcuts.id": "text def: nullable:NO",
    "paramsets.param_key": "text def: nullable:NO",
    "paramsets.param_value": "text def: nullable:NO",
    "paramsets.tile_number": "bigint def: nullable:NO",
    "postings.key_value": "text def: nullable:NO",
    "postings.tile_number": "bigint def: nullable:NO",
    "postings.trace_id": "bytea def: nullable:NO",
    "regressions.alert_id": "bigint def: nullable:NO",
    "regressions.commit_number": "bigint def: nullable:NO",
    "regressions.migrated": "boolean def: nullable:YES",
    "regressions.regression": "text def: nullable:YES",
    "regressions.regression_id": "text def: nullable:YES",
    "regressions2.alert_id": "bigint def: nullable:YES",
    "regressions2.cluster_summary": "jsonb def: nullable:YES",
    "regressions2.cluster_type": "text def: nullable:YES",
    "regressions2.commit_number": "bigint def: nullable:YES",
    "regressions2.creation_time": "timestamp with time zone def:now():::TIMESTAMPTZ nullable:YES",
    "regressions2.frame": "jsonb def: nullable:YES",
    "regressions2.id": "uuid def:gen_random_uuid() nullable:NO",
    "regressions2.is_improvement": "boolean def: nullable:YES",
    "regressions2.median_after": "real def: nullable:YES",
    "regressions2.median_before": "real def: nullable:YES",
    "regressions2.prev_commit_number": "bigint def: nullable:YES",
    "regressions2.triage_message": "text def: nullable:YES",
    "regressions2.triage_status": "text def: nullable:YES",
    "shortcuts.id": "text def: nullable:NO",
    "shortcuts.trace_ids": "text def: nullable:YES",
    "sourcefiles.source_file": "text def: nullable:NO",
    "sourcefiles.source_file_id": "bigint def:unique_rowid() nullable:NO",
    "subscriptions.bug_cc_emails": "ARRAY def: nullable:YES",
    "subscriptions.bug_component": "text def: nullable:YES",
    "subscriptions.bug_labels": "ARRAY def: nullable:YES",
    "subscriptions.bug_priority": "bigint def: nullable:YES",
    "subscriptions.bug_severity": "bigint def: nullable:YES",
    "subscriptions.contact_email": "text def: nullable:YES",
    "subscriptions.hotlists": "ARRAY def: nullable:YES",
    "subscriptions.name": "text def: nullable:NO",
    "subscriptions.revision": "text def: nullable:NO",
    "tracevalues.commit_number": "bigint def: nullable:NO",
    "tracevalues.source_file_id": "bigint def: nullable:YES",
    "tracevalues.trace_id": "bytea def: nullable:NO",
    "tracevalues.val": "real def: nullable:YES"
  },
  "IndexNames": [
    "commits.commits_git_hash_key",
    "culprits.by_revision",
    "favorites.by_user_id",
    "paramsets.by_tile_number",
    "postings.by_trace_id",
    "postings.by_key_value",
    "regressions2.by_commit_alert",
    "regressions2.by_alert_id",
    "sourcefiles.sourcefiles_source_file_key",
    "sourcefiles.by_source_file",
    "subscriptions.subscriptions_name_key",
    "tracevalues.by_source_file_id"
  ]
}
