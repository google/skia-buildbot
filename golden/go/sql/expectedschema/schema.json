{
  "ColumnNameAndType": {
    "changelists.changelist_id": "text def: nullable:NO",
    "changelists.last_ingested_data": "timestamp with time zone def: nullable:NO",
    "changelists.owner_email": "text def: nullable:NO",
    "changelists.status": "text def: nullable:NO",
    "changelists.subject": "text def: nullable:NO",
    "changelists.system": "text def: nullable:NO",
    "commitswithdata.commit_id": "text def: nullable:NO",
    "commitswithdata.tile_id": "integer def: nullable:NO",
    "deprecatedexpectationundos.expectation_id": "text def: nullable:NO",
    "deprecatedexpectationundos.id": "bigint def:unique_rowid() nullable:NO",
    "deprecatedexpectationundos.ts": "timestamp with time zone def: nullable:NO",
    "deprecatedexpectationundos.user_id": "text def: nullable:NO",
    "deprecatedingestedfiles.last_ingested": "timestamp with time zone def: nullable:NO",
    "deprecatedingestedfiles.source_file": "text def: nullable:NO",
    "deprecatedingestedfiles.source_file_id": "bytea def: nullable:NO",
    "diffmetrics.combined_metric": "real def: nullable:NO",
    "diffmetrics.dimensions_differ": "boolean def: nullable:NO",
    "diffmetrics.left_digest": "bytea def: nullable:NO",
    "diffmetrics.max_channel_diff": "smallint def: nullable:NO",
    "diffmetrics.max_rgba_diffs": "ARRAY def: nullable:NO",
    "diffmetrics.num_pixels_diff": "integer def: nullable:NO",
    "diffmetrics.percent_pixels_diff": "real def: nullable:NO",
    "diffmetrics.right_digest": "bytea def: nullable:NO",
    "diffmetrics.ts": "timestamp with time zone def: nullable:NO",
    "expectationdeltas.digest": "bytea def: nullable:NO",
    "expectationdeltas.expectation_record_id": "uuid def: nullable:NO",
    "expectationdeltas.grouping_id": "bytea def: nullable:NO",
    "expectationdeltas.label_after": "character def: nullable:NO",
    "expectationdeltas.label_before": "character def: nullable:NO",
    "expectationrecords.branch_name": "text def: nullable:YES",
    "expectationrecords.expectation_record_id": "uuid def:gen_random_uuid() nullable:NO",
    "expectationrecords.num_changes": "integer def: nullable:NO",
    "expectationrecords.triage_time": "timestamp with time zone def: nullable:NO",
    "expectationrecords.user_name": "text def: nullable:NO",
    "expectations.digest": "bytea def: nullable:NO",
    "expectations.expectation_record_id": "uuid def: nullable:YES",
    "expectations.grouping_id": "bytea def: nullable:NO",
    "expectations.label": "character def: nullable:NO",
    "gitcommits.author_email": "text def: nullable:NO",
    "gitcommits.commit_id": "text def: nullable:NO",
    "gitcommits.commit_time": "timestamp with time zone def: nullable:NO",
    "gitcommits.git_hash": "text def: nullable:NO",
    "gitcommits.subject": "text def: nullable:NO",
    "groupings.grouping_id": "bytea def: nullable:NO",
    "groupings.keys": "jsonb def: nullable:NO",
    "ignorerules.creator_email": "text def: nullable:NO",
    "ignorerules.expires": "timestamp with time zone def: nullable:NO",
    "ignorerules.ignore_rule_id": "uuid def:gen_random_uuid() nullable:NO",
    "ignorerules.note": "text def: nullable:YES",
    "ignorerules.query": "jsonb def: nullable:NO",
    "ignorerules.updated_email": "text def: nullable:NO",
    "metadatacommits.commit_id": "text def: nullable:NO",
    "metadatacommits.commit_metadata": "text def: nullable:NO",
    "options.keys": "jsonb def: nullable:NO",
    "options.options_id": "bytea def: nullable:NO",
    "patchsets.changelist_id": "text def: nullable:NO",
    "patchsets.commented_on_cl": "boolean def: nullable:NO",
    "patchsets.created_ts": "timestamp with time zone def: nullable:YES",
    "patchsets.git_hash": "text def: nullable:NO",
    "patchsets.patchset_id": "text def: nullable:NO",
    "patchsets.ps_order": "smallint def: nullable:NO",
    "patchsets.system": "text def: nullable:NO",
    "primarybranchdiffcalculationwork.calculation_lease_ends": "timestamp with time zone def: nullable:NO",
    "primarybranchdiffcalculationwork.grouping_id": "bytea def: nullable:NO",
    "primarybranchdiffcalculationwork.last_calculated_ts": "timestamp with time zone def: nullable:NO",
    "primarybranchparams.key": "text def: nullable:NO",
    "primarybranchparams.tile_id": "integer def: nullable:NO",
    "primarybranchparams.value": "text def: nullable:NO",
    "problemimages.digest": "text def: nullable:NO",
    "problemimages.error_ts": "timestamp with time zone def: nullable:NO",
    "problemimages.latest_error": "text def: nullable:NO",
    "problemimages.num_errors": "smallint def: nullable:NO",
    "secondarybranchdiffcalculationwork.branch_name": "text def: nullable:NO",
    "secondarybranchdiffcalculationwork.calculation_lease_ends": "timestamp with time zone def: nullable:NO",
    "secondarybranchdiffcalculationwork.digests": "ARRAY def: nullable:NO",
    "secondarybranchdiffcalculationwork.grouping_id": "bytea def: nullable:NO",
    "secondarybranchdiffcalculationwork.last_calculated_ts": "timestamp with time zone def: nullable:NO",
    "secondarybranchdiffcalculationwork.last_updated_ts": "timestamp with time zone def: nullable:NO",
    "secondarybranchexpectations.branch_name": "text def: nullable:NO",
    "secondarybranchexpectations.digest": "bytea def: nullable:NO",
    "secondarybranchexpectations.expectation_record_id": "uuid def: nullable:NO",
    "secondarybranchexpectations.grouping_id": "bytea def: nullable:NO",
    "secondarybranchexpectations.label": "character def: nullable:NO",
    "secondarybranchparams.branch_name": "text def: nullable:NO",
    "secondarybranchparams.key": "text def: nullable:NO",
    "secondarybranchparams.value": "text def: nullable:NO",
    "secondarybranchparams.version_name": "text def: nullable:NO",
    "secondarybranchvalues.branch_name": "text def: nullable:NO",
    "secondarybranchvalues.digest": "bytea def: nullable:NO",
    "secondarybranchvalues.grouping_id": "bytea def: nullable:NO",
    "secondarybranchvalues.options_id": "bytea def: nullable:NO",
    "secondarybranchvalues.secondary_branch_trace_id": "bytea def: nullable:NO",
    "secondarybranchvalues.source_file_id": "bytea def: nullable:NO",
    "secondarybranchvalues.tryjob_id": "text def: nullable:YES",
    "secondarybranchvalues.version_name": "text def: nullable:NO",
    "sourcefiles.last_ingested": "timestamp with time zone def: nullable:NO",
    "sourcefiles.source_file": "text def: nullable:NO",
    "sourcefiles.source_file_id": "bytea def: nullable:NO",
    "tiledtracedigests.digest": "bytea def: nullable:NO",
    "tiledtracedigests.grouping_id": "bytea def: nullable:NO",
    "tiledtracedigests.tile_id": "integer def: nullable:NO",
    "tiledtracedigests.trace_id": "bytea def: nullable:NO",
    "traces.corpus": "text def: nullable:NO",
    "traces.grouping_id": "bytea def: nullable:NO",
    "traces.keys": "jsonb def: nullable:NO",
    "traces.matches_any_ignore_rule": "boolean def: nullable:YES",
    "traces.trace_id": "bytea def: nullable:NO",
    "tracevalues.commit_id": "text def: nullable:NO",
    "tracevalues.digest": "bytea def: nullable:NO",
    "tracevalues.grouping_id": "bytea def: nullable:NO",
    "tracevalues.options_id": "bytea def: nullable:NO",
    "tracevalues.shard": "smallint def: nullable:NO",
    "tracevalues.source_file_id": "bytea def: nullable:NO",
    "tracevalues.trace_id": "bytea def: nullable:NO",
    "trackingcommits.last_git_hash": "text def: nullable:NO",
    "trackingcommits.repo": "text def: nullable:NO",
    "tryjobs.changelist_id": "text def: nullable:NO",
    "tryjobs.display_name": "text def: nullable:NO",
    "tryjobs.last_ingested_data": "timestamp with time zone def: nullable:NO",
    "tryjobs.patchset_id": "text def: nullable:NO",
    "tryjobs.system": "text def: nullable:NO",
    "tryjobs.tryjob_id": "text def: nullable:NO",
    "valuesathead.corpus": "text def: nullable:NO",
    "valuesathead.digest": "bytea def: nullable:NO",
    "valuesathead.grouping_id": "bytea def: nullable:NO",
    "valuesathead.keys": "jsonb def: nullable:NO",
    "valuesathead.matches_any_ignore_rule": "boolean def: nullable:YES",
    "valuesathead.most_recent_commit_id": "text def: nullable:NO",
    "valuesathead.options_id": "bytea def: nullable:NO",
    "valuesathead.trace_id": "bytea def: nullable:NO"
  },
  "IndexNames": [
    "changelists.system_status_ingested_idx",
    "changelists.status_ingested_idx",
    "expectationrecords.branch_ts_idx",
    "expectations.label_idx",
    "gitcommits.commit_idx",
    "patchsets.cl_order_idx",
    "primarybranchdiffcalculationwork.calculated_idx",
    "secondarybranchdiffcalculationwork.calculated_idx",
    "tiledtracedigests.tile_trace_idx",
    "tiledtracedigests.grouping_digest_idx",
    "tracevalues.trace_commit_idx",
    "traces.keys_idx",
    "traces.ignored_grouping_idx",
    "traces.grouping_ignored_idx",
    "tryjobs.cl_idx",
    "valuesathead.keys_idx",
    "valuesathead.ignored_grouping_idx",
    "valuesathead.corpus_commit_ignore_idx"
  ]
}
